controllers/operator/configmap.go

```diff
@@ -834,7 +834,7 @@ func (r *AuthenticationReconciler) generateCNCFClusterInfo(ctx context.Context,
 	clusterAddress := strings.Join([]string{strings.Join([]string{"cp-console", authCR.Namespace}, "-"), domainName}, ".")
 	clusterEndpoint := "https://" + clusterAddress
 	clusterAddressAuth := clusterAddress
-	if authCR.Spec.Config.ZenFrontDoor && ctrlcommon.ClusterHasZenExtensionGroupVersion(&r.DiscoveryClient) {
+	if shouldUseCPDHost(authCR, &r.DiscoveryClient) {
```

Comment:
I just happened to notice this could be replaced by the function I had defined in routes.go, so I went ahead and did that so that it's being reused where appropriate.